index.js

```diff
@@ -386,6 +410,122 @@ exports.getLastThreadPost = function (thread) {
   return msg
 }
 
+
+/* post revision utils */
+
+exports.getRevisions = function(ssb, thread, callback) {
+  function collectRevisions(thread, callback) {
+    // this function walks the revisions of a given thread, collecting them up
+    // asyncly
+    var deepThread = pluckRecursively(thread, 'related')[0]
+
+    var threadRevisionLogCB = multicb({pluck: 1})
+    deepThread
+    .filter(function(relatedMsg) { return isaRevisionTo(relatedMsg, thread) })
+    .forEach(function(edit) {
+      exports.createRevisionLog(ssb, edit, threadRevisionLogCB())
+    })
+    
+    threadRevisionLogCB(function(err, revLogs) {
+      if (err) callback(err)
+      const connectedLogs = revLogs
+        .filter(function(revLog) { 
+          // filter down to the log that links to this thread, by finding the log
+          // whose messages point to this thread as root 
+          // we should only need to test one message if createRevisionLog is
+          // passing tests
+          return revLog[revLog.length-1].key === thread.key
+        })
+
+      // the longest such log is the one that contains all of the revisions
+      const logLengths = connectedLogs.map(function(log) { return log.length })
+      const thisLog = connectedLogs.find(function(log) {
+                        return log.length === Math.max.apply(Math, logLengths)
+                      })
+      callback(null, thisLog)
+    })
+    
+  }
+            
+
+  if (!thread.hasOwnProperty('related')) { 
+    // if the thread doesn't have its related objects,fetch them
+    ssb.relatedMessages({id: thread.key, count: true}, function(err, enrichedThread) {
+      if (err) callback (err)
+      // if still no related objects
+      else if (!enrichedThread.hasOwnProperty('related')) callback(null, [])
+      else collectRevisions(enrichedThread, callback)
+    })
+  } else {
+    collectRevisions(thread, callback)
+  }  
+}
+
+exports.getLatestRevision = function (ssb, thread, callback) {  
+  var msg = thread
+  if (!msg.hasOwnProperty('related')) {
+    ssb.relatedMessages({id: msg.key, count: true}, function(err, richMsg) {
+      if (err) callback (err)
+      // if still no related objects
+      else if (!richMsg.hasOwnProperty('related')) { callback(null, richMsg) }
+      else { traverseRelatedEdits(richMsg, callback) }
+    })
+  } else if (!msg.hasOwnProperty('related')) {
+    callback(null, msg)
+  } else {
+    traverseRelatedEdits(msg, callback)
+  }
```

Comment:
i tested it again and i remember why this was such a pain. if you try to get the related edits from a message that isn't far enough down the tree, it only tells you about the ones before it. i don't remember how i thought it worked anymore, but to see this, try running the tests with a version of this function that uses the method you suggest. a couple of the edge tests written specifically for this will fail.

```js
exports.getLatestRevision = function (ssb, thread, callback) {
  const sortRevsGetLast = function(richMsg) {
    const sortedRevs = richMsg.related
            .filter(function (rev) {
              return rev.value.content.type === 'post-edit'
            })
            .sort(function(msgA, msgB) {
              return msgA.value.sequence < msgB.value.sequence
            })                         
    return sortedRevs[0]
  }
  var msg = thread
  if (!msg.hasOwnProperty('related')) {
    ssb.relatedMessages({id: msg.key, count: true}, function(err, richMsg) {
      if (err) callback (err)
      // if still no related objects
      else if (!richMsg.hasOwnProperty('related')) { callback(null, richMsg) }
      else {
        callback(null, sortRevsGetLast(richMsg))
      }
      // else { traverseRelatedEdits(richMsg, callback) }
    })
  } else if (!msg.hasOwnProperty('related')) {
    callback(null, msg)
  } else {
//    traverseRelatedEdits(msg, callback)
    callback(null, sortRevsGetLast(msg))
  }
}
```